pkg/ruler/base/api.go

```diff
@@ -356,77 +356,151 @@ func parseGroupName(params map[string]string) (string, error) {
 	return groupName, nil
 }
 
+type Request struct {
+	UserID    string
+	Namespace string
+	Group     string
+	Labels    []labels.Label
+}
+
 // parseRequest parses the incoming request to parse out the userID, rules namespace, and rule group name
 // and returns them in that order. It also allows users to require a namespace or group name and return
 // an error if it they can not be parsed.
-func parseRequest(req *http.Request, requireNamespace, requireGroup bool) (string, string, string, error) {
+func parseRequest(req *http.Request, requireNamespace, requireGroup bool) (Request, error) {
 	userID, err := tenant.TenantID(req.Context())
 	if err != nil {
-		return "", "", "", user.ErrNoOrgID
+		return Request{}, user.ErrNoOrgID
+	}
+
+	err = req.ParseForm()
+	if err != nil {
+		return Request{}, err
 	}
 
 	vars := mux.Vars(req)
 
 	namespace, err := parseNamespace(vars)
 	if err != nil {
 		if err != ErrNoNamespace || requireNamespace {
-			return "", "", "", err
+			return Request{}, err
 		}
 	}
 
 	group, err := parseGroupName(vars)
 	if err != nil {
 		if err != ErrNoGroupName || requireGroup {
-			return "", "", "", err
+			return Request{}, err
+		}
+	}
+
+	var ls []labels.Label
+	ql := req.Form.Get("labels")
+	if ql != "" {
+		labelStrings := strings.Split(ql, ",")
+		ls = make([]labels.Label, len(labelStrings))
+
+		for i, l := range labelStrings {
+			kv := strings.Split(l, ":")
+			if len(kv) != 0 {
+				ls[i] = labels.Label{
+					Name:  kv[0],
+					Value: kv[1],
+				}
+			}
```

Comment:
Can you please split this out into its own function so that we can unit-test it independently?